com.regnosys.rosetta/src/com/regnosys/rosetta/types/Rosetta.xsemantics

```diff
@@ -983,21 +990,25 @@ checkrule CheckOnlyExistsExpression for RosettaOnlyExistsExpression e from {
 			}
 		}
 	}
-	if (parent === null && !e.hasImplicitParent) {
+	if (parent === null && !e.implicitVariableExistsInContext) {
 		fail error "Object must have a parent object."
 			source e.args.head
 	}
 	
-	var Data parentData = null
+	var RListType parentType = null
 	if (parent !== null) {
 		looseConstraintCheck(parent, single)
-		empty |- parent : var RListType pt
-		if (pt !== null) {
-			parentData = (pt.itemType as RDataType).^data
-		}
+		empty |- parent : parentType
 	} else {
-		parentData = e.implicitParent
+		parentType = e.typeOfImplicitVariable.orElse(null)
+		if (parentType !== null) {
+			single.isSubconstraintOf(parentType.constraint)
+			or
+			fail error notConstraintMessage(single, parentType)
+				source e
+		}
```

Comment:
In conditions, it was already possible to leave out the parent of attributes when performing a `only exists` operation, e.g., `a, b, c only exists` if `a`, `b` and `c` are attributes of the enclosing data type.

However, until now this was impossible to do in the context of a functional expression, e.g., `inputs extract [ a, b, c only exists ]`. This is fixed now.

Reply:
Nice